### PR TITLE
Fix #11

### LINKED ISSUE #11: Possible solution regex problem in generic parser
I think we can solve this issue by deleting"." and "," in the string. I tried a little bit and it seems working normal and no problem with 4 digits prices.

I tried changing part of code to this;
```
t = strings.ReplaceAll(t, ".", "")
t = strings.ReplaceAll(t, ",", "")
price, err := strconv.ParseInt(t, 10, 64)
if err != nil {
	return nil, fmt.Errorf("error transforming string to integer (from %v): %v", t, err)
}

return money.New(price, currency), nil
```

Original code;
https://github.com/ahmetb/baklava/blob/1dbca945f30cbe4e585ba4905f9d9ecad8f37e35/providers/genericparser/base.go#L64C1-L77C49

### PR DESCRIPTION
**Pull request by eminks**
I resolved the issue in #11 and wrote a unit test.

### CODE REVIEW

File: providers/genericparser/base.go [hunk 1]
```diff
@@ -44,35 +44,31 @@ func (_ GenericParser) FromURL(selector, url string) (*money.Money, error) {
 	if t == "" {
 		return nil, fmt.Errorf("selector didn't match anything")
 	}
-	t = strings.TrimFunc(t, unicode.IsSpace)
-	t = strings.ReplaceAll(t, "\n", "")
 
 	currency := "TRY" // assume default
-	if strings.HasSuffix(t, "TL") {
+	if regexp.MustCompile(`\b(tl|try|₺)\b`).MatchString(strings.ToLower(t)) {
```
Comment (ahmetb) [inline providers/genericparser/base.go:49]: This prob won't match TL uppercase?

File: providers/genericparser/base_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "test/mockTestPage.html")
+	})
+
+	// Create a new mock server
+	server := httptest.NewServer(handler)
+	defer server.Close()
+	price_one, err := genericparser.GenericParser{}.FromURL(`#price_one`, server.URL)
+	assert.Nil(t, err)
+	assert.Equal(t, price_one.Display(), "₺1,045.00", "The two price should be the same.")
```
Comment (ahmetb) [inline providers/genericparser/base_test.go]: No need for that last string.

File: providers/genericparser/base_test.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+func TestGenericParser(t *testing.T) {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "test/mockTestPage.html")
+	})
+
+	// Create a new mock server
+	server := httptest.NewServer(handler)
+	defer server.Close()
+	price_one, err := genericparser.GenericParser{}.FromURL(`#price_one`, server.URL)
```
Comment (ahmetb) [inline providers/genericparser/base_test.go]: Go variable naming conventions

File: providers/genericparser/base.go (hunk 1 above)
Comment (eminks) [reply providers/genericparser/base.go:49]: Before matching process string gets through strings.ToLower function. Also unit tests include uppercase TL

**Pull request merged**